programs/stake/src/stake_state.rs

```diff
@@ -182,17 +182,23 @@ fn redeem_stake_rewards(
         inflation_point_calc_tracer.as_ref(),
         credits_auto_rewind,
     )
-    .map(|(stakers_reward, voters_reward, credits_observed)| {
-        if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer {
-            inflation_point_calc_tracer(&InflationPointCalculationEvent::CreditsObserved(
-                stake.credits_observed,
-                Some(credits_observed),
-            ));
-        }
-        stake.credits_observed = credits_observed;
-        stake.delegation.stake += stakers_reward;
-        (stakers_reward, voters_reward)
-    })
+    .map(
+        |StakeRewardDistributions {
+             staker_rewards,
+             voter_rewards,
+             credits_observed,
+         }| {
```

Comment:
nits: this might be personal taste, but i'd like to see a more nuanced rewrite than this verbatim version like below:

```rust
.map(|stake_reward_distributions| {
  ...
  stale.credits_observed = stake_reward_distributions.credits_observed;
  ...
})
```

(or destructure `stake_reward_distributions`if this looks too wordy, java-esque... xD)

imo, large destructuring in params causes some reading pauses and makes it easy to forget the context of unqualified identifiers like `credits_observed` after that.